packages/ia-components/sandbox/details/AnchorSearch.jsx

```diff
@@ -63,21 +63,24 @@ class AnchorSearch extends IAReactComponent {
 
   constructor(props) {
     super(props); // { query || field & value, sort, reload }
-    //TODO-STATE this might have the issue of constructor not being re-run and needing componentDidMount catch
-    this.setState({
-      query: this.props.query || ObjectFromEntries([[this.props.field, this.props.value]]),
-      urlProps: ObjectFilter(this.props, (k, unusedV) => AnchorSearch.urlparms.includes(k)),  //sort, reload, query
-      anchorProps: ObjectFilter(this.props, (k, unusedV) => (!AnchorSearch.urlparms.includes(k) && !['children'].includes(k)))
-    });
+    this.onClick = this.onClick.bind(this);
+    this.state = {};
   }
 
-  clickCallable(unusedEvent) {
+  static getDerivedStateFromProps(props, unusedState) {
+    return {
+      query: props.query || ObjectFromEntries([[props.field, props.value]]),
+      urlProps: ObjectFilter(props, (k, unusedV) => AnchorSearch.urlparms.includes(k)),  //sort, reload, query
+      anchorProps: ObjectFilter(props, (k, unusedV) => (!AnchorSearch.urlparms.includes(k) && !['children'].includes(k)))
+    };
+  }
+  onClick(ev) {
     // Note this is only called in dweb; !Dweb has a director href
     debug('Clicking on link to search: %s', this.state.query);
     DwebArchive.Nav.navSearch(
       this.state.query,
       { noCache: this.props.reload, wanthistory: !this.props.reload, sort: this.props.sort });
-    return false; // Dont propagate event
+    ev.preventDefault(); // Dont propogate event
```

Comment:
commenting on this as i am not sure preventing events to flow freely is the right way to go.

Reply:
Same answer as for AnchorDownload - Dweb only, and otherwise its going to trigger the href. (And its useful to have the href there for things like the display in the status bar and so on)